fe/fe-core/src/main/java/com/starrocks/analysis/Expr.java

```diff
@@ -372,6 +375,102 @@ public void analyzeNoThrow(Analyzer analyzer) {
         }
     }
 
+    /**
```

Comment:
We have been removing functions from classes that define types, and these functions are not recommended to be placed in Expr. If you think AnalyzerUtils.java is inappropriate, you can extract the logic to other locations separately

Reply:
In the first place, I add extractConjunct(Expr) and compoundAnd(Expr) to AnalyzerUtils.java accroding to extractConjunct(ScalarOperator) and compoundAnd(ScalarOperator) are written in Utils.java. But there seems a lot of utility functions defined in the Expr, so I think it's better to put them together. And we can refactor in another pr, moving those utility functions to Utils or other appropriate places.